### ISSUE TITLE
parallelFor interrupt when found a solution

### ISSUE THREAD

**Issue by privefl** [RcppCore/RcppParallel#108]
I'm trying to implement an algorithm that tests a condition over many elements in a loop.
When one condition if false, then I return false (like function `all()` basically).

Is there any way to tell the other threads to stop when one has found a condition that is false?

**Comment by eddelbuettel** [comment]
I don't have a tip handy but this strikes me a common problem within parallel computing. With MPI you could probably pass a "we are done" message along.  TBB likely has something but I don't know what it is.

**Comment by kevinushey** [comment]
There's some discussion around how this can be made possible using a `task_group_context`:

https://software.intel.com/en-us/node/506312

https://www.threadingbuildingblocks.org/docs/help/tbb_userguide/Cancellation_and_Nested_Parallelism.html

That basically gives you a handle on a set of tasks, which can then be cancelled in bulk. It might be worth exploring that if you're willing to write your own TBB-specific code.

**Comment by privefl** [comment]
Thanks for the tips.
I found a somewhat working solution.
I'm using a local boolean to see if a solution has been found.
With a grain size of 1, the first thread to find a solution gets to finish the loop very quickly.
But, this requires waiting for the other threads to finish the one iteration they started.

Actually, I also tried to use OpenMP which has some cancellation signals. Yet, this is always much slower than using the solution I present above.
